tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/ProtoConverter.java

```diff
@@ -251,22 +264,26 @@ protected Expr visit(ColumnRef node, Object context) {
     // the first executor of a DAG request.
     IntegerCodec.writeLong(cdo, position);
     builder.setVal(cdo.toByteString());
+
+    builder.setFieldType(toPBFieldType(getType(node)));
     return builder.build();
   }
 
   @Override
   protected Expr visit(Constant node, Object context) {
     Expr.Builder builder = Expr.newBuilder();
+    DataType type = node.getType();
     if (node.getValue() == null) {
       builder.setTp(ExprType.Null);
-      return builder.build();
     } else {
-      DataType type = node.getType();
       builder.setTp(type.getProtoExprType());
       CodecDataOutput cdo = new CodecDataOutput();
       type.encode(cdo, EncodeType.PROTO, node.getValue());
       builder.setVal(cdo.toByteString());
     }
+    if (type.getType() == MySQLType.TypeTimestamp) {
+      type = DataTypeFactory.of(MySQLType.TypeDatetime);
```

Comment:
very nice! can you add some comments for these lines? (or provide an issue link perhaps?)

Reply:
This is some code I forget to remove. :(